libs/LighthouseConvert.js

```diff
@@ -0,0 +1,115 @@
+    const text = data.toString();
+    let lines = _.filter(text.split(/(.+)+/g), (line) => line !== '' && line.length > 5); // Removing empty elements
+
+    // Creating series
+    const series = _.map(lines, (line, index) => {
+        const config = line.split(':');
+        return {
+            id: index + 1,
+            abbr: config[0],
+            type: config[1],
+            unit: config[2],
+            precision: config[3],
+            name: config[4],
+            source: config[5],
+            fields: config[6],
+            convert: config[2] ? Number(config[2].replace(/[^0-9\.]+/g, '')) : 1
+        };
+    });
+
+    // Write to file
+    await fsp.writeFile('./data/series.json', JSON.stringify(series)).catch(error => console.error(error));
+
+    return series;
+}
+
+const lhc = {
+    /** 
+     * Retrieves all series in json format
+     * @returns {Array<Object>} an array of objects representing all the series
+    */
+    getSeries: () => {
+        return seriesJson;
+    },
+
+    /**
+     * Converts an array of observations given by a sx### table
+     * @param {string} seriesKey either the name or abbreviation of the series
+     * @param {Array<number>} values an array of values that will be converted
+     * @returns {Array<number>} an array of numbers that were converted to the proper value
+     * @throws {Error} serie is null or undefined, cannot be found
+     */
+    convertObservations: (seriesKey, values) => {
+        const serie = lhc.findSerie(seriesKey);
+        
+        if (serie == null || serie == undefined) {
+            throw new Error(`Cannot find the series provided by ${seriesKey}`);
+        }
+
+        return _.map(values, (value) => {
+            if (serie.convert && serie.convert === 0) {
+                // do nothing
+            } else {
+                return +(value * serie.convert).toPrecision(default_percision);
+            }
+        });
+    },
+
+    /**
+     * Gets the default percision for all numbers being converted
+     * @returns {number} the default_percision
+     */
+    getDefaultPercision: () => {
+        return default_percision;
+    },
+
+    /**
+     * Sets the default percision for all numbers being converted
+     * @param {number} newPercision a number that determines the number of decimal places
+     * @throws {Error} newPrecision is not a valid number
+     */
+    setDefaultPercision: (newPercision) => {
+        if (!isNan(newPercision)) {
+            default_percision = newPercision;
+        } else {
+            throw new Error('newPrecison is an invalid number');
+        }
+    },
+
+    /**
+     * Looks for a series in the series list defined in series.json
+     * @param {string} searchValue a string that will be searched on the series abbr and name
+     * @returns {Object} a serie that is found in series
+     * @throws {Error} search value is null or undefined
+     */
+    findSerie: (searchValue) => {
```

Comment:
I don't think Serie is a word. You can name it findOneSeries, findASeries, or findSingleSeries